binderhub/main.py

```diff
@@ -51,6 +73,66 @@ def get(self, provider_prefix, spec):
         )
 
 
+
+class SettingsHandler(BaseHandler):
+    """
+    This shows the settings page, mostly get the know binder for current user
+    from a cookie, and display it as a list. Also get the default binder
+    selected by the user and mark it as such in the settings UI.
+    """
+
+    def get(self, arg):
+        return self._get()
+    
+    def _get(self, cookie=None):
+        binders = cookie or self.get_json_cookie('known_binders')
+        self.add_header('X-Frame-Options','DENY')
+        self.render_template(
+            "settings.html", 
+            binders=binders['known'],
+            default_binder= binders['default'],
+            raw=json.dumps(binders, indent=2),
+            enable_federation_portal=self.settings['enable_federation_portal'],
+            google_analytics_code=self.settings['google_analytics_code'],
+            google_analytics_domain=self.settings['google_analytics_domain'],
+        )
+
+    def post(self, arg):
+        nd = self.get_body_argument('default')
+        cookie = self.get_json_cookie('known_binders')
+        cookie['default'] = nd
+        cookie = self._normalize_cookie(cookie)
+        self.set_json_cookie('known_binders', cookie)
```

Comment:
We'll want to set `secure: True` if served over HTTPS, and `httponly=True`

Reply:
See [JupyterHub's set_cookie](https://github.com/jupyterhub/jupyterhub/blob/646c853cf4025896d5990d57e8991174b40dab19/jupyterhub/handlers/base.py#L300) for an example.